pkg/pool-utils/contracts/NewBasePool.sol

```diff
@@ -214,13 +214,17 @@ abstract contract NewBasePool is
 
         _ensureNotPaused();
         if (totalSupply() == 0) {
-            (bptAmountOut, amountsIn) = _onInitializePool(sender, userData);
+            (bptAmountOut, amountsIn) = _onInitializePool(sender, recipient, userData);
 
             // On initialization, we lock _getMinimumBpt() by minting it for the zero address. This BPT acts as a
             // minimum as it will never be burned, which reduces potential issues with rounding, and also prevents the
             // Pool from ever being fully drained.
+            // Some pool types do not require this mechanism, and the minimum BPT might be zero.
             _require(bptAmountOut >= _getMinimumBpt(), Errors.MINIMUM_BPT);
-            _mintPoolTokens(address(0), _getMinimumBpt());
+            if (_getMinimumBpt() > 0) {
+                _mintPoolTokens(address(0), _getMinimumBpt());
+            }
```

Comment:
I'd remove this since we're considering making the minimum be non-zero. This is another point of extra complexity.